### DISCUSSION TITLE
Razzle for production

### DISCUSSION THREAD

**Discussion by mbuguanewton** [jaredpalmer/razzle#1588] [Q&A] [answered]
Razzle has been great so far. But I have only used it for small apps. I would like to know if devs would rewrite their whole codebase to use razzle for a production application. Basically how safe is it to use razzle for a production app?

**Comment by fivethreeo** [comment]
https://github.com/jaredpalmer/razzle/issues/332#issuecomment-778455917

**Comment by squiddy** [comment, marked as answer]
That likely depends on your starting point, do you think you need to rewrite your whole codebase?

In our case, we're starting with a home-grown solution which is quite messy. We were building the same code once with babel for the server, and with webpack for the client. Due to that, most nice webpack features were out of question because we were also doing server-side rendering.

A while ago we discussed our options:

* moving to razzle, which is more of an evolution of our approach
* moving to NextJS

---

Moving to razzle (still ongoing) has been pretty easy so far, our application code itself needed no adjustments whatsoever. We just needed some smaller changes to our express server (serving of assets, adjustments to loadable).

Moving our custom webpack configuration over to razzle was a bit tricky in one case, because we were using multiple entrypoints (no SPA yet).

---

On the other hand, we tried moving to NextJS, but just the routing alone was causing us problems. NextJS has much more conventions than razzle, so it would have required lots of changes to our application code.

---

Our razzle-powered website is not yet running in production, but I don't foresee any issues there. Razzle is more like a library and get's out of our way. In the worst-case scenario - razzle development stops - we could still "eject" out of it, and still be in a much, much better place than we were before.
